File: examples/python/testing-patterns/basic-tests/callback-function.py
from aws_durable_execution_sdk_python import DurableContext, durable_execution
from aws_durable_execution_sdk_python.config import CallbackConfig

@durable_execution
def handler(event: dict, context: DurableContext) -> str:
    callback_config = CallbackConfig(
        timeout_seconds=120,
        heartbeat_timeout_seconds=60
    )
    
    callback = context.create_callback(
        name="example_callback",
        config=callback_config
    )
    
    return f"Callback created with ID: {callback.callback_id}"

File: examples/java/testing-patterns/basic-tests/callback-function.java
public class CallbackHandler extends DurableHandler<Map, String> {
    @Override
    public String handleRequest(Map input, DurableContext ctx) {
        var callbackConfig = CallbackConfig.builder()
            .timeout(Duration.ofSeconds(120))
            .heartbeatTimeout(Duration.ofSeconds(60))
            .build();

        var callback = ctx.createCallback("example_callback", String.class, callbackConfig);
        return "Callback created with ID: " + callback.getCallbackId();
    }
}
